snuba/cli/consumer.py

```diff
@@ -188,3 +188,5 @@ def handler(signum: int, frame: Any) -> None:
     signal.signal(signal.SIGTERM, handler)
 
     consumer.run()
+    consumer_builder.commit_log_producer.close()
+    consumer_builder.replacements_producer.close()
```

Comment:
These are internal details of the consumer builder. Why don't you add a public close method there that in turn calls close on the producers?